packages/blocks/src/api/registration.js

```diff
@@ -844,12 +848,21 @@ export const registerBlockBindingsSource = ( source ) => {
 	}
 
 	// Check the `label` property is correct.
-	if ( ! label ) {
+	if ( label && existingSource?.label ) {
```

Comment:
Is the intention to prevent the registration of the source on the client if the label gets defined in both places?

The API is still private so that might be fine, but I have mixed feelings about such strict verification. What is the reasoning behind it?

Reply:
You're right that maybe we should not prevent registration and just throw a warning. I can make that change. Do you think this should be the case for other use cases? For example, when any of the properties is not a valid value?

Reply:
Too many labels seems harmless, but it seems good to inform the dev which one was selected. In other cases, if they provide a wrong value and it’s something optional we probably could also skip it and proceed. Well, unless you want to always prevent registration then no changes are needed. Maybe being very strict is easier strategy to have 🤔

Reply:
My line of thinking right now is that too many labels won't "break" anything. But for example, if I add an invalid `setValues` or `canUserEditValue`, it will probably break the block when you try to edit it. For that reason, maybe it is safer to keep it restricted. Unless until we receive more feedback from users after using the API.

Reply:
With the invalid option setting provided, we would have to ignore them. It sounds right to wait and leave it unchanged. Thank you for discussing and discarding the alternative together with me 😄